#HG001

/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/multiIntersectBed -i /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/GIABv3.3.2/GRCh38_remapped_HG001_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-X_v.3.3.2_highconf_RTGandPGphasetransfer_comphetindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/GIABv3.3.2/GRCh38_remapped_HG001_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-X_v.3.3.2_highconf_RTGandPGphasetransfer_comphetsnp10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/GIABv3.3.2/GRCh38_remapped_HG001_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-X_v.3.3.2_highconf_RTGandPGphasetransfer_complexindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/GIABv3.3.2/GRCh38_remapped_HG001_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-X_v.3.3.2_highconf_RTGandPGphasetransfer_snpswithin10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/PG2016-1.0/GRCh38_remapped_PG2016-1.0_NA12878_b37_comphetindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/PG2016-1.0/GRCh38_remapped_PG2016-1.0_NA12878_b37_comphetsnp10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/PG2016-1.0/GRCh38_remapped_PG2016-1.0_NA12878_b37_complexindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/PG2016-1.0/GRCh38_remapped_PG2016-1.0_NA12878_b37_snpswithin10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/RTG_37.7.3/GRCh38_remapped_sp_v37.7.3.NA12878_comphetindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/RTG_37.7.3/GRCh38_remapped_sp_v37.7.3.NA12878_comphetsnp10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/RTG_37.7.3/GRCh38_remapped_sp_v37.7.3.NA12878_complexindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/RTG_37.7.3/GRCh38_remapped_sp_v37.7.3.NA12878_snpswithin10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/SVs/GRCh38_remapped_PacBio_MetaSV_mergedSVs.bed.gz  | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin | /Volumes/DroboZook/bioinfo/htslib-1.3.2/bgzip -c > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG001_NA12878/GIABv3.3.2/GRCh38_remapped_HG001_genomespecific_complexandSVs_v3.3.2_PG_RTG.bed.gz



#HG002

#find union of complex variants and SVs for 3.3.2
/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/multiIntersectBed -i /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv3.3.2/GRCh38_remapped_HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_triophased_comphetindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv3.3.2/GRCh38_remapped_HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_triophased_comphetsnp10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv3.3.2/GRCh38_remapped_HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_triophased_complexindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv3.3.2/GRCh38_remapped_HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_triophased_snpswithin10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/SVs/GRCh38_remapped_HG002_HG003_HG004_allsvs_merged.bed.gz | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin | /Volumes/DroboZook/bioinfo/htslib-1.3.2/bgzip -c > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv3.3.2/GRCh38_remapped_HG002_genomespecific_complexandSVs_v3.3.2.bed.gz

##HG002 v4.1
#Merge nearby phased variants to create the subsets below (vcflib sometimes outputs extra duplicate lines, so need to sort and deduplicate)
/Volumes/DroboZook/bioinfo/vcflib_140404/bin/vcfgeno2haplo -w 10 -r /Volumes/giab/data/reference_genomes/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set.fna /Volumes/giab/analysis/v4-development/NIST_v4.1_SmallVariantDraftBenchmark_12182019/GRCh38/HG002_GRCh38_1_22_v4.1_draft_benchmark.vcf  > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/intermediatefiles/HG002_GRCh38_1_22_v4.1_draft_benchmark_geno2haplo10.vcf

#Find compound hets with no length change (i.e., only compound hets where both alleles contain only SNPs or MNPs within 10bp) 
grep '1/2\|1|2\|2|1\|2/1' /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/intermediatefiles/HG002_GRCh38_1_22_v4.1_draft_benchmark_geno2haplo10.vcf | awk 'BEGIN {FS = "\t\|,"; OFS = "\t"} {if (length($4)==length($5) && length($4)==length($6)) print $1,$2-50,$2+length($4)+50}' | sed 's/^chr//;s/^X/23/' | sort -k1,1n -k2,2n | sed 's/^23/X/;s/^/chr/' | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin -d 50 > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_comphetsnp10bp_slop50.bed

/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/intersectBed -a /Volumes/giab/analysis/v4-development/NIST_v4.1_SmallVariantDraftBenchmark_12182019/GRCh38/HG002_GRCh38_1_22_v4.1_draft_benchmark.vcf -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_comphetsnp10bp_slop50.bed | wc -l
#   30475
/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/intersectBed -a /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/intermediatefiles/HG002_GRCh38_1_22_v4.1_draft_benchmark_geno2haplo10.vcf -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_comphetsnp10bp_slop50.bed | cut -f1-5 | sort -k1,1 -k2,2n | uniq | wc -l
#   18278

#Find compound hets where at least 1 allele has a length change (i.e., only compound hets where at least one allele contains an indel within 10bp) 
grep '1/2\|1|2\|2|1\|2/1' /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/intermediatefiles/HG002_GRCh38_1_22_v4.1_draft_benchmark_geno2haplo10.vcf | awk 'BEGIN {FS = "\t\|,"; OFS = "\t"} {if (length($4)!=length($5) || length($4)!=length($6)) print $1,$2-50,$2+length($4)+50}' | sed 's/^chr//;s/^X/23/' | sort -k1,1n -k2,2n | sed 's/^23/X/;s/^/chr/' | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin -d 50 > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_comphetindel10bp_slop50.bed

/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/intersectBed -a /Volumes/giab/analysis/v4-development/NIST_v4.1_SmallVariantDraftBenchmark_12182019/GRCh38/HG002_GRCh38_1_22_v4.1_draft_benchmark.vcf -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_comphetindel10bp_slop50.bed | wc -l
#   65257
/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/intersectBed -a /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/intermediatefiles/HG002_GRCh38_1_22_v4.1_draft_benchmark_geno2haplo10.vcf -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_comphetindel10bp_slop50.bed | cut -f1-5 | sort -k1,1 -k2,2n | uniq | wc -l
#   62227

#Find complex variants that are not compound hets where there is a length change and it is not a simple insertion or deletion
grep -v '1/2\|1|2\|2|1\|2/1' /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/intermediatefiles/HG002_GRCh38_1_22_v4.1_draft_benchmark_geno2haplo10.vcf | awk 'BEGIN {FS = "\t"; OFS = "\t"} {if (length($4)!=length($5) && length($4)>1 && length($5)>1) print $1,$2-50,$2+length($4)+50}' | sed 's/^chr//;s/^X/23/' | sort -k1,1n -k2,2n | sed 's/^23/X/;s/^/chr/' | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin -d 50 > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_complexindel10bp_slop50.bed

/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/intersectBed -a /Volumes/giab/analysis/v4-development/NIST_v4.1_SmallVariantDraftBenchmark_12182019/GRCh38/HG002_GRCh38_1_22_v4.1_draft_benchmark.vcf -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_complexindel10bp_slop50.bed | wc -l
#   42098
/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/intersectBed -a /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/intermediatefiles/HG002_GRCh38_1_22_v4.1_draft_benchmark_geno2haplo10.vcf -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_complexindel10bp_slop50.bed | cut -f1-5 | sort -k1,1 -k2,2n | uniq | wc -l
#   23625

#Find SNPs within 10bp that are not compound hets
grep -v '^#\|1/2\|1|2\|2|1\|2/1' /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/intermediatefiles/HG002_GRCh38_1_22_v4.1_draft_benchmark_geno2haplo10.vcf | awk 'BEGIN {FS = "\t"; OFS = "\t"} {if (length($4)==length($5) && length($4)>1 ) print $1,$2-50,$2+length($4)+50}' | sed 's/^chr//;s/^X/23/' | sort -k1,1n -k2,2n | sed 's/^23/X/;s/^/chr/' | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin -d 50 > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_snpswithin10bp_slop50.bed

/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/intersectBed -a /Volumes/giab/analysis/v4-development/NIST_v4.1_SmallVariantDraftBenchmark_12182019/GRCh38/HG002_GRCh38_1_22_v4.1_draft_benchmark.vcf -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_snpswithin10bp_slop50.bed | wc -l
#   227708
/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/intersectBed -a /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/intermediatefiles/HG002_GRCh38_1_22_v4.1_draft_benchmark_geno2haplo10.vcf -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_snpswithin10bp_slop50.bed | cut -f1-5 | sort -k1,1 -k2,2n | uniq | wc -l
#   101929

#Find additional unphased complex variants and overlapping variants that vcflib ignores
awk '{FS = OFS = "\t"} { print $1,$2,$2+length($4)}' /Volumes/giab/analysis/v4-development/NIST_v4.1_SmallVariantDraftBenchmark_12182019/GRCh38/HG002_GRCh38_1_22_v4.1_draft_benchmark.vcf | grep -v ^# | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin -d 10 -c 3 -o count | awk '{FS = OFS = "\t"} { if($4>1) print $1,$2-50,$3+50}' | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/subtractBed -a stdin -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_comphetindel10bp_slop50.bed | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/subtractBed -a stdin -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_comphetsnp10bp_slop50.bed | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/subtractBed -a stdin -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_complexindel10bp_slop50.bed | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/subtractBed -a stdin -b /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_snpswithin10bp_slop50.bed > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_othercomplexwithin10bp_slop50.bed

#find union of CNVs and SVs
/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/multiIntersectBed -i /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/SVs/expanded_150_GRCh38_remapped_HG002_SVs_Tier1plusTier2_v0.6.1.bed /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/SVs/GRCh38_mrcanavar_intersect_ccs_1000_window_size_cnv_threshold_intersect_ont_1000_window_size_cnv_threshold.bed /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/SVs/GRCh38_remapped_HG002_SVs_Tier1plusTier2_v0.6.1.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/SVs/GRCh38_union_HG002_CCS_15kb_20kb_merged_ONT_1000_window_size_combined_elliptical_outlier_threshold.bed /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/SVs/HG2_SKor_TrioONTCanu_intersect_HG2_SKor_TrioONTFlye_intersect_HG2_SKor_CCS15_gt10kb_GRCh38.bed /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/SVs/SVMergeInversions.GRCh38.120519.clustered_slop150_chr1_22.bed | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin | /Volumes/DroboZook/bioinfo/htslib-1.3.2/bgzip -c > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/SVs/GRCh38_HG002_genomespecific_CNVsandSVs_v4.1.bed.gz
#find union of complex variants and SVs
/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/multiIntersectBed -i /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_comphetindel10bp_slop50.bed /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_comphetsnp10bp_slop50.bed /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_complexindel10bp_slop50.bed /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_snpswithin10bp_slop50.bed /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_GRCh38_1_22_v4.1_draft_benchmark_othercomplexwithin10bp_slop50.bed /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/SVs/GRCh38_HG002_genomespecific_CNVsandSVs_v4.1.bed.gz | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin | /Volumes/DroboZook/bioinfo/htslib-1.3.2/bgzip -c > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG002/GIABv4.1/HG002_genomespecific_complexandSVs_v4.1.bed.gz


##HG003 v3.3.2

#find union of complex variants and SVs
/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/multiIntersectBed -i /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG003/GIABv3.3.2/GRCh38_remapped_HG003_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X_CHROM1-22_v.3.3.2_highconf_comphetindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG003/GIABv3.3.2/GRCh38_remapped_HG003_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X_CHROM1-22_v.3.3.2_highconf_comphetsnp10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG003/GIABv3.3.2/GRCh38_remapped_HG003_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X_CHROM1-22_v.3.3.2_highconf_complexindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG003/GIABv3.3.2/GRCh38_remapped_HG003_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X_CHROM1-22_v.3.3.2_highconf_snpswithin10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG003/SVs/GRCh38_remapped_HG002_HG003_HG004_allsvs_merged.bed.gz | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin | /Volumes/DroboZook/bioinfo/htslib-1.3.2/bgzip -c > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG003/GIABv3.3.2/GRCh38_remapped_HG003_genomespecific_complexandSVs_v3.3.2.bed.gz


##HG004 v3.3.2

#find union of complex variants and SVs
/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/multiIntersectBed -i /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG004/GIABv3.3.2/GRCh38_remapped_HG004_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X_CHROM1-22_v.3.3.2_highconf_comphetindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG004/GIABv3.3.2/GRCh38_remapped_HG004_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X_CHROM1-22_v.3.3.2_highconf_comphetsnp10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG004/GIABv3.3.2/GRCh38_remapped_HG004_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X_CHROM1-22_v.3.3.2_highconf_complexindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG004/GIABv3.3.2/GRCh38_remapped_HG004_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X_CHROM1-22_v.3.3.2_highconf_snpswithin10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG004/SVs/GRCh38_remapped_HG002_HG003_HG004_allsvs_merged.bed.gz  | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin | /Volumes/DroboZook/bioinfo/htslib-1.3.2/bgzip -c > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG004/GIABv3.3.2/GRCh38_remapped_HG004_genomespecific_complexandSVs_v3.3.2.bed.gz


##HG005 v3.3.2


#find union of complex variants and SVs
/Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/multiIntersectBed -i /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG005/GIABv3.3.2/GRCh38_remapped_HG005_GRCh37_highconf_CG-IllFB-IllGATKHC-Ion-SOLID_CHROM1-22_v.3.3.2_highconf_comphetindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG005/GIABv3.3.2/GRCh38_remapped_HG005_GRCh37_highconf_CG-IllFB-IllGATKHC-Ion-SOLID_CHROM1-22_v.3.3.2_highconf_comphetsnp10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG005/GIABv3.3.2/GRCh38_remapped_HG005_GRCh37_highconf_CG-IllFB-IllGATKHC-Ion-SOLID_CHROM1-22_v.3.3.2_highconf_complexindel10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG005/GIABv3.3.2/GRCh38_remapped_HG005_GRCh37_highconf_CG-IllFB-IllGATKHC-Ion-SOLID_CHROM1-22_v.3.3.2_highconf_snpswithin10bp_slop50.bed.gz /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG005/SVs/GRCh38_remapped_HG005_HG006_HG007_FB_GATKHC_CG_MetaSV_allsvs_merged.bed.gz  | /Volumes/DroboZook/bioinfo/bedtools2.25.0/bin/mergeBed -i stdin | /Volumes/DroboZook/bioinfo/htslib-1.3.2/bgzip -c > /Volumes/giab/analysis/GRCh38_stratification_bed_files/GenomeSpecific/HG005/GIABv3.3.2/GRCh38_remapped_HG005_genomespecific_complexandSVs_v3.3.2.bed.gz
